### ISSUE TITLE
http not listening on set IP address

### ISSUE THREAD

**Issue by jpkelly** [bitfocus/companion#358]
Maybe not a bug but user error..
Starting arm7I build #1087 with `./headless.js 10.0.0.42 8000` does not set http listening to 10.0.0.42

On a working build I saw in the output:
```
$./headless.js 10.0.0.42 8000
...
get_userconfig_all { page_direction_flipped: false,
  artnet_enabled: false,
  artnet_universe: '1',
  artnet_channel: '1' }
  lib/db db_get(bank) +206ms
  lib/db db_get(instance) +0ms
  lib/http new url: http://10.0.0.42:8000/ +0ms
  config config written +11s
  config config written +1ms
  lib/db db_save written +160ms
  lib/db db_save renamed +2ms
...
```

With a new build I see the following : (the `lib/http new url: http://10.0.0.42:8000/` line is missing)
```
$./headless.js 10.0.0.42 8000
...
get_userconfig_all { page_direction_flipped: false,
  artnet_enabled: false,
  artnet_universe: '1',
  artnet_channel: '1' }
  lib/db db_get(bank) +198ms
  lib/db db_get(instance) +1ms
  lib/db db_save written +153ms
  lib/db db_save renamed +2ms
  lib/update update server says {} +11s
  lib/update fresh data received {} +2ms
...
```

**Comment by jpkelly** [comment]
Here is the full output on startup:
```
$ ./headless.js 10.0.0.42 8000
  app configuration directory /home/pi +0ms
  app mkdirp /home/pi/companion/ null +21ms
sef { http_port: 8000, bind_ip: '127.0.0.1', start_minimised: false }
  lib/db new(db) +0ms
  lib/db db loaded +1ms
  lib/db db_get(userconfig) +4ms
  lib/update loading update +0ms
  lib/update read uuid S1YLx2OAQ +9ms
  lib/db db_get(page) +32ms
  lib/db db_get(feedbacks) +143ms
  lib/db db_get(bank) +9ms
  lib/elgato_dm module required +0ms
  lib/elgato_dm add device emulator +15ms
  lib/elgato_emulator Adding Elgato Streamdeck Emulator +0ms
  lib/device loading for emulator +0ms
  lib/db db_get(deviceconfig) +8s
  lib/bank bank-update saving +0ms
  lib/db db_set(bank, [object Object]) +170ms
  lib/db db_save begin +2ms
  lib/graphics Generating buffers +0ms
  lib/graphics Done +997ms
  lib/elgato_dm USB: checking devices (blocking call) +1s
  lib/elgato_dm add device 0001:0005:00 +21ms
  lib/elgato_dm USB: done +31ms
  lib/db db_get(bank_actions) +1s
  lib/db db_get(bank_release_actions) +1ms
  lib/action got instance +0ms
  lib/db db_get(instance) +13ms
  serialport:binding:auto-detect loading LinuxBinding +0ms
get_userconfig_all { page_direction_flipped: false,
  artnet_enabled: false,
  artnet_universe: '1',
  artnet_channel: '1' }
  lib/db db_get(bank) +194ms
  lib/db db_get(instance) +1ms
  lib/db db_save written +152ms
  lib/db db_save renamed +2ms
  lib/update update server says {} +11s
  lib/update fresh data received {} +1ms
  lib/instance instance_init { 'bitfocus-companion': 
   { instance_type: 'bitfocus-companion',
     label: 'internal',
     id: 'bitfocus-companion' } } +0ms
  lib/instance loaded module 7thsensedelta@1.0.0 by Per Røine <per.roine@gmail.com> +14ms
  lib/instance loaded module aja-helo@1.0.0 by Casey Selph <cxselph@gmail.com> +7ms
  lib/instance loaded module amp@1.0.0 by Håkon Nessjøen <haakon@bitfocus.io> +5ms
  lib/instance loaded module analogway-eks500@1.0.0 by Adrian Davis <themusicnerd@gmail.com> +11ms
  lib/instance loaded module analogway-midra@1.0.0 by Dorian Meid <meid@backstage.org> +8ms
  lib/instance loaded module analogway-pls300@1.0.1 by Tyler Krupa +7ms
  lib/instance loaded module artnet@1.0.0 by Håkon Nessjøen <haakon@bitfocus.io> +11ms
  lib/instance loaded module atem@1.0.1 by Håkon Nessjøen <haakon@bitfocus.io> +434ms
  lib/instance loaded module avplayback@1.0.0 by Per Røine <per.roine@gmail.com> +4ms
  lib/instance loaded module barco-encore@1.0.0 by William Viker <william@bitfocus.io> +5ms
  lib/instance loaded module bitfocus-companion@1.0.1 by Håkon Nessjøen <haakon@bitfocus.io> +4ms
  lib/instance loaded module bmd-smartview@1.0.0 by Per Roine <per.roine@gmail.com> +5ms
  lib/instance loaded module casparcg@1.0.0 by Håkon Nessjøen <haakon@bitfocus.io> +8ms
  lib/instance loaded module chamsys@1.0.0 by Per Røine <per.roine@gmail.com> +4ms
  lib/instance loaded module chamsys-magicqudp@1.0.0 by Per Røine <per.roine@gmail.com> +4ms
  lib/instance loaded module christiepj@1.0.0 by Per Røine <per.roine@gmail.com> +6ms
  lib/instance loaded module cockos-reaper@1.0.0 by Oliver Herman <oliver@monoxane.com> +4ms
  lib/instance loaded module dcs@1.0.0 by Per Røine <per.roine@gmail.com> +16ms
  lib/instance loaded module disguise@1.0.0 by William Viker <william@bitfocus.io> +4ms
  lib/instance loaded module disguise-mtc@1.0.0 by William Viker <william@bitfocus.io> +4ms
  lib/instance loaded module dphighlight@1.0.0 by Per Røine <per.roine@gmail.com> +4ms
  lib/instance loaded module eventmaster@1.0.0 by William Viker <william@bitfocus.io> +157ms
  lib/instance loaded module generic-http@1.0.0 by William Viker <william@bitfocus.io> +3ms
  lib/instance loaded module globalcache-itac-cc@1.0.0 by Casey Selph <cxselph@gmail.com> +5ms
  lib/instance loaded module globalcache-itac-ir@1.0.0 by Casey Selph <cxselph@gmail.com> +5ms
  lib/instance loaded module globalcache-itac-sl@1.0.0 by Casey Selph <cxselph@gmail.com> +4ms
  lib/instance loaded module grandma2@1.0.0 by William Viker <william.viker@gmail.com> +17ms
  lib/instance loaded module highend-hog4@1.0.0 by Per Røine <per.roine@gmail.com> +4ms
  lib/instance loaded module horae@1.0.0 by Daniel Richert <daniel.richert@svv.fi> +4ms
  lib/instance loaded module hyperdeck@1.0.0 by Per Roine <per.roine@gmail.com> +5ms
  lib/instance loaded module imagepro@1.0.0 by Per Røine <per.roine@gmail.com> +5ms
  lib/instance loaded module irisdown-countdowntimer@1.0.0 by Per Røine <per.roine@gmail.com> +5ms
  lib/instance loaded module kipro@1.0.0 by Per Røine <per.roine@gmail.com> +4ms
  lib/instance loaded module kramer-vp727@1.0.0 by Per Røine <per.roine@gmail.com> +4ms
  lib/instance loaded module lightware2@1.0.1 by Håkon Nessjøen <haakon@bitfocus.io> +7ms
  lib/instance loaded module lightware3@1.0.0 by Håkon Nessjøen <haakon@bitfocus.io> +6ms
  lib/instance loaded module livecore@1.0.0 by Dorian Meid <meid@backstage.org> +8ms
  lib/instance loaded module millumin@1.0.0 by Per Røine <per.roine@gmail.com> +5ms
  lib/instance loaded module mitti@1.0.0 by Per Røine <per.roine@gmail.com> +5ms
  lib/instance loaded module modulo@1.0.0 by Per Røine <per.roine@gmail.com> +5ms
  lib/instance loaded module motu-avb@1.0.0 by Per Røine <per.roine@gmail.com> +5ms
  lib/instance loaded module nevion-mrp@1.0.0 by William Viker <william.viker@gmail.com> +6ms
common
  lib/instance loaded module obs-studio@1.0.0 by William Viker <william@bitfocus.io> +927ms
  lib/instance loaded module octopus@1.0.0 by Jeffrey Davidsz <jeffrey.davidsz@vicreo.eu> +6ms
  lib/instance loaded module octopusapp@1.0.0 by Per Røine <per.roine@gmail.com> +4ms
  lib/instance loaded module osc@1.0.0 by William Viker <william@bitfocus.io> +4ms
  lib/instance loaded module panasonic-avhs@1.0.0 by Håkon Nessjøen <haakon@bitfocus.io> +7ms
  lib/instance loaded module pds@1.0.0 by Dorian Meid <meid@backstage.org> +6ms
  lib/instance loaded module pjlink@1.0.0 by Per Roine <per.roine@gmail.com> +4ms
  lib/instance loaded module playbackproplus@1.0.0 by Per Røine <per.roine@gmail.com> +7ms
  lib/instance loaded module pptrsc@1.0.0 by Per Røine <per.roine@gmail.com> +4ms
  lib/instance loaded module ptzoptics-visca@1.0.0 by Håkon Nessjøen <haakon@bitfocus.io> +10ms
  lib/instance loaded module pvp@1.0.0 by Per Røine <per.roine@gmail.com> +4ms
  lib/instance loaded module qlab@1.0.0 by Per Røine <per.roine@gmail.com> +5ms
  lib/instance loaded module renewedvision-propresenter@2.1.0 by Oliver Herrmann <oliver@monoxane.com> +5ms
  lib/instance loaded module resolume-arena@1.0.0 by Oliver Herman <oliver@monoxane.com> +4ms
  lib/instance loaded module rosstalk@1.0.0 by William Viker <william@bitfocus.io> +4ms
  lib/instance loaded module scs@1.0.0 by Per Røine <per.roine@gmail.com +4ms
  lib/instance loaded module sony-visca@1.2.1 by Per Røine <per.roine@gmail.com> +12ms
  lib/instance loaded module spyder@1.0.0 by Per Røine <per.roine@gmail.com> +5ms
  lib/instance loaded module studiocoast-vmix@1.0.0 by Per Røine <per.roine@gmail.com> +4ms
  lib/instance loaded module tascamcd@1.0.0 by Håkon Nessjøen <haakon@bitfocus.io> +4ms
  lib/instance loaded module videohub@1.0.2 by William Viker <william@bitfocus.io> +6ms
  lib/instance loaded module vio@1.0.0 by Dorian Meid <meid@backstage.org> +6ms
  lib/instance loaded module vlc@1.0.0 by Håkon Nessjøen <haakon@bitfocus.io> +4ms
  lib/instance loaded module vyvphoton@1.0.0 by Per Roine <per.roine@gmail.com> +3ms
  lib/instance loaded module watchout-production@1.0.0 by Dorian Meid <meid@backstage.org> +5ms
  lib/instance loaded module x32@1.0.0 by Per Roine <per.roine@gmail.com> +6ms
  lib/instance loaded module xair@1.0.0 by Per Roine <per.roine@gmail.com> +7ms
  lib/instance getting instance_get_package_info +2ms
  instance:bitfocus-companion:bitfocus-companion instance save +0ms
  lib/db db_set(instance, [object Object]) +4s
  lib/db db_save begin +1ms
  lib/instance Running _init of bitfocus-companion +4ms
  lib/action adding action bitfocus-companion:instance_control +4s
  lib/action adding action bitfocus-companion:set_page +0ms
  lib/action adding action bitfocus-companion:panic +1ms
  lib/usb got local variables from module +0ms
  lib/usb module added successfully BJrb_2dR7 +0ms
  lib/elgato_dm initializing deviceHandler +4s
  lib/device loading for 0001:0005:00 +5s
  lib/db db_get(deviceconfig) +18ms
  lib/device Device AL37G1A06967 was on page 2 +1ms
  lib/bank bank-update saving +5s
  lib/db db_set(bank, [object Object]) +1ms
  lib/action adding action bitfocus-companion:instance_control +13ms
  lib/action adding action bitfocus-companion:set_page +0ms
  lib/action adding action bitfocus-companion:panic +0ms
  lib/usb/elgato elgato.prototype.begin() +0ms
  lib/db db_save written +236ms
  lib/db db_save renamed +1ms
  lib/elgato_dm remove device 0001:0005:00 +1s
  lib/device unloading for 0001:0005:00 +1s
  lib/action adding action bitfocus-companion:instance_control +1s
  lib/action adding action bitfocus-companion:set_page +0ms
  lib/action adding action bitfocus-companion:panic +1ms
  lib/usb/elgato elgato.prototype.clearDeck() +2s
  lib/db db_save begin +4s
  lib/db db_save written +4ms
  lib/db db_save renamed +1ms
```

**Comment by jpkelly** [comment]
Running `sudo ./headless.js 10.0.0.42 8000` does work.

**Comment by jpkelly** [comment]
Seems to be fixed in build #1109

**Comment by haakonnessjoen** [comment]
Great!
